packages/ember-template-compiler/lib/plugins/deprecate-view-and-controller-paths.js

```diff
@@ -0,0 +1,60 @@
+function DeprecateViewAndControllerPaths() {
+  // set later within HTMLBars to the syntax package
+  this.syntax = null;
+}
+
+/**
+  @private
+  @method transform
+  @param {AST} ast The AST to be transformed.
+*/
+DeprecateViewAndControllerPaths.prototype.transform = function DeprecateViewAndControllerPaths_transform(ast) {
+  if (!!Ember.ENV._ENABLE_LEGACY_VIEW_SUPPORT) {
+    return;
+  }
+  var walker = new this.syntax.Walker();
+
+  walker.visit(ast, function(node) {
+    if (!validate(node)) { return; }
+
+    deprecatePath(node.path);
+    deprecatePaths(node.params);
+    deprecateHash(node.hash);
+
+  });
+
+  return ast;
+};
+
+function deprecateHash(hash) {
+  if (!hash || !hash.pairs) {
+    return;
+  }
+  var i, l, pair, paths;
+  for (i=0, l=hash.pairs.length;i<l;i++) {
+    pair = hash.pairs[i];
+    paths = pair.value.params;
+    deprecatePaths(paths);
+  }
+}
+
+function deprecatePaths(paths) {
+  if (!paths) {
+    return;
+  }
+  var i, l, path;
+  for (i=0, l=paths.length;i<l;i++) {
+    path = paths[i];
+    deprecatePath(path);
+  }
+}
+
+function deprecatePath(path) {
+  Ember.deprecate(`Using {{${path && path.type === 'PathExpression' && path.parts[0]}}} or any path based on it has been deprecated. Please review the deprecation guide for further details.`, !(path && path.type === 'PathExpression' && (path.parts[0] === 'view' || path.parts[0] === 'controller')), { url: 'http://emberjs.com/deprecations/v1.x#toc_view-and-controller-template-keywords' });
```

Comment:
We need the location info in this message.  You can use https://github.com/emberjs/ember.js/blob/master/packages/ember-template-compiler/lib/system/calculate-location-display.js to build up the the same string as the other AST transforms.

Take a look at https://github.com/emberjs/ember.js/blob/master/packages/ember-template-compiler/lib/plugins/transform-input-on-to-onEvent.js#L50-L59 for an example of how you would interpolate it.